changelog.d/13830.bugfix

```diff
@@ -0,0 +1 @@
+typing: check origin server of typing event against servers currently in the room. We also use a method that do not block on partial state so the transaction doesn't get stuck during a fast join.
```

Comment:
In light of Rich's comment about making the changelog more understandable to users, maybe we can describe the impact of the bug in the changelog:
```suggestion
Fix a long-standing bug where typing events would be accepted from remote servers not present in a room. Also fix a bug where incoming typing events would cause other incoming events to get stuck during a fast join.
```